### PR TITLE
wrap image write in try

### PR DESCRIPTION
**Pull request by tripzero**
This sometimes fails due to "Unsupported format" error.
Report the exception and move on.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -110,7 +110,8 @@ def collect_files(self):
         self.multi_speaker = len(l) == 5
         texts = list(map(lambda l: l.decode("utf-8").split("|")[3], lines))
         if self.multi_speaker:
-            speaker_ids = list(map(lambda l: int(l.decode("utf-8").split("|")[-1]), lines))
+            speaker_ids = list(
```
Comment (r9y9) [inline train.py]: Could you revert unrelated changes?

File: train.py [hunk 2]
```diff
@@ -397,23 +404,31 @@ def eval_model(global_step, writer, device, model, checkpoint_dir, ismultispeake
...
-            tag = "eval_averaged_alignment_{}_{}".format(idx, speaker_str)
-            writer.add_image(tag, np.uint8(cm.viridis(np.flip(alignment, 1).T) * 255), global_step)
+            try:
+                path = join(eval_output_dir, "step{:09d}_text{}_{}_alignment.png".format(
+                    global_step, idx, speaker_str))
+                save_alignment(path, alignment)
+                tag = "eval_averaged_alignment_{}_{}".format(idx, speaker_str)
+                writer.add_image(tag, np.uint8(cm.viridis(
+                    np.flip(alignment, 1).T) * 255), global_step)
+            except Exception as e:
```
Comment (r9y9) [inline train.py]: Please keep `try: ... except` as minimal as possible. I think 

```py
try:
   writer.add_image(...)
```

is better.

File: train.py (hunk 1 above)
Comment (tripzero) [reply train.py]: yep. sorry, my editor auto-fixes too long lines

**Comment by tripzero** [comment]
Are CI failures related to the pytorch version?  What can we do to get deepvoice working on the latest?

**Comment by tripzero** [comment]
Are CI failures related to the pytorch version?  What can we do to get deepvoice working on the latest?

File: train.py [hunk 3]
```diff
@@ -442,44 +448,63 @@ def save_states(global_step, writer, mel_outputs, linear_outputs, attn, mel, y,
...
+        except Exception as e:
+            warn(str(e))
 
     # Predicted mel spectrogram
     if mel_outputs is not None:
-        mel_output = mel_outputs[idx].cpu().data.numpy()
-        mel_output = prepare_spec_image(audio._denormalize(mel_output))
-        writer.add_image("Predicted mel spectrogram", mel_output, global_step)
+        try:
+            mel_output = mel_outputs[idx].cpu().data.numpy()
```
Comment (r9y9) [inline train.py]: Could you move this line to the out of try block? Seems irrelevant.

File: train.py [hunk 4]
```diff
@@ -442,44 +448,63 @@ def save_states(global_step, writer, mel_outputs, linear_outputs, attn, mel, y,
...
-            alignment_dir = join(checkpoint_dir, "alignment_layer{}".format(i + 1))
+        try:
+            # Save averaged alignment
+            alignment_dir = join(checkpoint_dir, "alignment_ave")
             os.makedirs(alignment_dir, exist_ok=True)
-            path = join(alignment_dir, "step{:09d}_layer_{}_alignment.png".format(
-                global_step, i + 1))
+            path = join(alignment_dir,
+                        "step{:09d}_alignment.png".format(global_step))
+            alignment = attn.mean(0)[idx].cpu().data.numpy()
```
Comment (r9y9) [inline train.py]: Please move the code to out of try block.
